bigquery/google/cloud/bigquery/job.py

```diff
@@ -1253,53 +1389,52 @@ class QueryJob(_AsyncJob):
     :param client: A client which holds credentials and project configuration
                    for the dataset (which requires a project).
 
-    :type udf_resources: tuple
-    :param udf_resources: An iterable of
-                        :class:`google.cloud.bigquery._helpers.UDFResource`
-                        (empty by default)
-
-    :type query_parameters: tuple
-    :param query_parameters:
-        An iterable of
-        :class:`google.cloud.bigquery._helpers.AbstractQueryParameter`
-        (empty by default)
-
     :type job_config: :class:`~google.cloud.bigquery.job.QueryJobConfig`
     :param job_config:
         (Optional) Extra configuration options for the query job.
     """
     _JOB_TYPE = 'query'
     _UDF_KEY = 'userDefinedFunctionResources'
-    _QUERY_PARAMETERS_KEY = 'queryParameters'
 
-    def __init__(self, job_id, query, client,
-                 udf_resources=(), query_parameters=(), job_config=None):
+    def __init__(self, job_id, query, client, job_config=None):
         super(QueryJob, self).__init__(job_id, client)
 
         if job_config is None:
             job_config = QueryJobConfig()
+        if job_config.use_legacy_sql is None:
+            job_config.use_legacy_sql = False
 
         self.query = query
-        self.udf_resources = udf_resources
-        self.query_parameters = query_parameters
         self._configuration = job_config
         self._query_results = None
 
     @property
-    def use_legacy_sql(self):
+    def allow_large_results(self):
```

Comment:
I thought the plan was to remove all these delegations from the job class? Looks like you removed them then added them back.

Reply:
I removed the setters but not the getters. We could potentially make `job.configuration` public if we wanted to get rid of the getters, but I wasn't sure it was worth the extra layer of indirection.

Reply:
SGTM. I will add getters to LoadJob.